Apps/W1/HybridGP/app/src/pages/GPMigrationConfiguration.Page.al

```diff
@@ -0,0 +1,512 @@
+    begin
+        if SettingsHasCompanyMissingDimension() then
+            if (not Confirm(CompanyMissingDimensionExitQst)) then
+                exit(false);
+
+        if ShowManagementPromptOnClose then
+            if Confirm(OpenCloudMigrationPageQst) then
+                Page.Run(page::"Intelligent Cloud Management");
+
+        exit(true);
+    end;
+
+    local procedure SettingsHasCompanyMissingDimension(): Boolean
+    var
+        GPCompanyAdditionalSettingsCompanies: Record "GP Company Additional Settings";
+    begin
+        GPCompanyAdditionalSettingsCompanies.SetFilter("Name", '<>%1', '');
+        GPCompanyAdditionalSettingsCompanies.FindSet();
+
+        repeat
+            if (GPCompanyAdditionalSettingsCompanies."Global Dimension 1" = '') then
+                exit(true);
+
+            if (GPCompanyAdditionalSettingsCompanies."Global Dimension 2" = '') then
+                exit(true);
+
+        until GPCompanyAdditionalSettingsCompanies.Next() = 0;
+
+        exit(false);
+    end;
+
+    local procedure AssignDimension(DimensionNumber: Integer; DimensionLabel: Text[30])
+    var
+        GPCompanyAdditionalSettingsCompanies: Record "GP Company Additional Settings";
+    begin
+        GPCompanyAdditionalSettingsCompanies.SetFilter("Name", '<>%1', '');
+        if GPCompanyAdditionalSettingsCompanies.FindSet() then begin
+            repeat
+                if (DimensionLabel = '') or CompanyHasSegment(GPCompanyAdditionalSettingsCompanies.Name, DimensionLabel) then begin
+                    if DimensionNumber = 1 then
+                        GPCompanyAdditionalSettingsCompanies.Validate("Global Dimension 1", DimensionLabel);
+
+                    if DimensionNumber = 2 then
+                        GPCompanyAdditionalSettingsCompanies.Validate("Global Dimension 2", DimensionLabel);
+
+                    GPCompanyAdditionalSettingsCompanies.Modify();
+                end;
+            until GPCompanyAdditionalSettingsCompanies.Next() = 0;
+        end;
+    end;
+
+    local procedure CompanyHasSegment(CompanyName: Text[50]; SegmentName: Text[30]): Boolean
+    var
+        GPSegmentName: Record "GP Segment Name";
+    begin
+        GPSegmentName.SetRange("Company Name", CompanyName);
+        GPSegmentName.SetRange("Segment Name", SegmentName);
+
+        exit(GPSegmentName.FindFirst());
+    end;
+
+    var
+        GPCompanyAdditionalSettings: Record "GP Company Additional Settings";
+        ShowIntroductionNotification: Boolean;
+        ShowManagementPromptOnClose: Boolean;
+        IntroNotificationMsg: Label 'Use this configuration page to specify what information will be migrated from GP to Business Central.', Locked = true;
+        CompanyMissingDimensionExitQst: Label 'A Company is missing a Dimension. Are you sure you want to exit?', Locked = true;
+        OpenCloudMigrationPageQst: Label 'Would you like to open the Cloud Migration Management page to manage your data migrations?', Locked = true;
```

Comment:
These labels should not be locked, we should translate them if needed

Reply:
Updated